engine/src/flutter/impeller/renderer/backend/vulkan/sampler_library_vk.h

```diff
@@ -20,13 +21,16 @@ class SamplerLibraryVK final
   // |SamplerLibrary|
   ~SamplerLibraryVK() override;
 
+  explicit SamplerLibraryVK(const std::weak_ptr<DeviceHolderVK>& device_holder);
+
+  void SetWorkarounds(const WorkaroundsVK& workarounds);
```

Comment:
Nit: instead of having a setter you could provide workarounds in the constructor and then then could be const instance variables.

Reply:
There is sort of a chicken and egg here because we create the sampler library before we do the workarounds. Its probably possible to untangle but its a huge PITA